apiserver/facades/controller/caasapplicationprovisioner/provisioner.go

```diff
@@ -603,32 +603,35 @@ func (a *API) Units(ctx context.Context, args params.Entities) (params.CAASUnits
 			results.Results[i].Error = apiservererrors.ServerError(err)
 			continue
 		}
-		app, err := a.state.Application(appName.Id())
-		if err != nil {
+		appId, err := a.applicationService.GetApplicationIDByName(ctx, appName.Id())
```

Comment:
Perhaps break lines 601 -> 634 into a new func. I'd say it does enough to warrant a new func, perhaps called something like "getUnitStatusesResult" and it'll read nicely like: 
```go
results.Results[i] = getUnitStatusesResult
```

Reply:
yeah fair enough